### PR TITLE
feat(Core/Logging): improve db logging

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
* Delete config option for enable db logging (for enable need db appender)
* Cleanup worldserver.conf
* Delete unused config options
* Fix skip db logging after `.reload conf`

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/5478
---

image

### CODE REVIEW

**Comment by Kitzunu** [comment]
Perhaps off-topic, but what would be the equivalent to SQLDriver with the new log system?

**Comment by Winfidonarleyan** [comment]
> Perhaps off-topic, but what would be the equivalent to SQLDriver with the new log system?

```ini
#Logger.sql.driver=4,Console Server
```

File: src/server/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -3207,32 +3188,37 @@ MoveMaps.Enable = 1
 Minigob.Manabonk.Enable = 1
 
 #
-
 #    Allow.IP.Based.Action.Logging
 #        Description: Logs actions, e.g. account login and logout to name a few, based on IP of current session.
 #        Default:     0 - (Disabled)
 #                     1 - (Enabled)
+#
```
Comment (Kitzunu) [inline src/server/worldserver/worldserver.conf.dist:3195]: Why add extra under? Shouldnt it only be one extra above?

File: src/server/worldserver/worldserver.conf.dist [hunk 2]
```diff
@@ -3388,6 +3374,7 @@ Appender.Console=1,4,0,"1 9 3 6 5 8"
 Appender.Server=2,5,0,Server.log,w
 Appender.GM=2,5,15,gm_%s.log
 Appender.DBErrors=2,5,0,DBErrors.log
+# Appender.DB=3,5,0
```
Comment (Kitzunu) [inline src/server/worldserver/worldserver.conf.dist:3377]: Should it be commented out?

**Comment by nicoaravena** [comment]
Little test with all `chatlogs.* = 1`  and `LogDB.Chat = 1`, nothing saved to database.

Also, type should be 10 (or LOG_TYPE_CHAT) based on wiki?

File: src/server/worldserver/worldserver.conf.dist (hunk 1 above)
Comment (Winfidonarleyan) [reply src/server/worldserver/worldserver.conf.dist:3195]: I don't know the standards and I always do this

File: src/server/worldserver/worldserver.conf.dist (hunk 2 above)
Comment (Winfidonarleyan) [reply src/server/worldserver/worldserver.conf.dist:3377]: Yes, db logging disabled default

**Comment by Winfidonarleyan** [comment]
> Little test with all `chatlogs.* = 1` and `LogDB.Chat = 1`, nothing saved to database.
> 
> Also, type should be 10 (or LOG_TYPE_CHAT) based on wiki?

I didn't understand anything.

**Comment by nicoaravena** [comment]
I can see that this settings are completely new and not documented in the wiki (just in the repo), can you mention the rework pull please or at least docs next time to avoid this kind of problems please. Will check again.

Also with the type, I based my comment on this https://www.azerothcore.org/wiki/logs (and of course this is outdated).

**Comment by nicoaravena** [comment]
Now thinks works fine.

imagen

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan please update the wiki so it's clear to everyone how to use this

**Comment by mpfans** [comment]
image

**Comment by mpfans** [comment]
It doesn't seem to support mysql8

**Comment by Winfidonarleyan** [comment]
https://github.com/azerothcore/wiki/pull/445

**Comment by mpfans** [comment]
image

**Comment by mpfans** [comment]
set
`Appender.DB=3,5,0`
`Logger.chat.log=5,DB`
`Logger.entities.player=5,DB`
`Logger.warden=5,DB`
and then
image

**Comment by mpfans** [comment]
image

File: src/server/game/World/World.cpp [hunk 3]
```diff
@@ -425,6 +425,9 @@ void World::LoadConfigSettings(bool reload)
         sLog->LoadFromConfig();
     }
 
+    if (uint32 realmId = sConfigMgr->GetOption<uint32>("RealmID", 0)) // 0 reserved for auth
+        sLog->SetRealmId(realmId);
```
Comment (ChromieCraftServer) [inline src/server/game/World/World.cpp:429]: add brackets please

File: src/server/worldserver/worldserver.conf.dist [hunk 4]
```diff
@@ -1804,14 +1793,6 @@ GM.InGMList.Level = 3
 
 GM.InWhoList.Level = 3
 
-#
-#    GM.LogTrade
-#        Description: Include GM trade and trade slot enchanting operations in GM log.
-#        Default:     1 - (Enabled)
-#                     0 - (Disabled)
-
-GM.LogTrade = 1
```
Comment (FrancescoBorzi) [inline src/server/worldserver/worldserver.conf.dist:1813]: why is this gone?

**Comment by FrancescoBorzi** [comment]
@mpfans can this be merged?

**Comment by mpfans** [comment]
add `Logger.chat.log.addon=4,DB`,and then no error... 

but can't record a lot of logs, like transactions, like loot, like ah.

**Comment by FrancescoBorzi** [comment]
> add `Logger.chat.log.addon=4,DB`,and then no error...
> 
> but can't record a lot of logs, like transactions, like loot, like ah.

@Winfidonarleyan

**Comment by Winfidonarleyan** [comment]
> add `Logger.chat.log.addon=4,DB`,and then no error...
> 
> but can't record a lot of logs, like transactions, like loot, like ah.

Transactions log only console and file type.
Now not all loggers works, need replace `"server"` loggers

**Comment by mpfans** [comment]
If you don't think about implementing other functions, I think you can merge them

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**